File: backend/app/schemas/report.py
from pydantic import BaseModel
from datetime import date, datetime


class ReportCreate(BaseModel):
    title: str
    period_start: date | None = None
    period_end: date | None = None
    status: str = "draft"


class ReportUpdate(BaseModel):
    title: str | None = None
    content_html: str | None = None
    period_start: date | None = None
    period_end: date | None = None
    status: str | None = None
    recipients: str | None = None


class ReportResponse(BaseModel):
    id: int
    title: str
    content_html: str | None
    period_start: date | None
    period_end: date | None
    status: str
    recipients: str | None
    sent_date: date | None
    ai_model_used: str | None
    project_id: int
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/report.ts
/**
 * Report type definitions.
 * Aligned with backend schemas: ReportResponse.
 */

/** Matches backend ReportResponse schema */
export interface Report {
  id: number;
  title: string;
  content_html: string | null;
  period_start: string | null;
  period_end: string | null;
  status: string;
  recipients: string | null;
  sent_date: string | null;
  ai_model_used: string | null;
  project_id: number;
  created_at: string;
}

/** Cross-project report view used in ReportsPage */
export interface ReportView {
  id: number;
  type: string;
  projectId: number;
  projectName: string;
  date: string;
  status: string;
  periodStart: string;
  periodEnd: string;
  recipients: string;
  aiGenerated: boolean;
  htmlContent: string;
}
